### PR TITLE
[SPARK-24090][K8S] Update running-on-kubernetes.md

### PR DESCRIPTION
**Pull request by liyinan926**
## What changes were proposed in this pull request?

Updated documentation for Spark on Kubernetes for the upcoming 2.4.0.

Please review http://spark.apache.org/contributing.html before opening a pull request.

@mccheah @erikerlandson

### CODE REVIEW

**Review by srowen — CHANGES_REQUESTED** [review]

File: docs/running-on-kubernetes.md [hunk 1]
```diff
@@ -299,21 +328,15 @@ RBAC authorization and how to configure Kubernetes service accounts for pods, pl
 
 ## Future Work
 
-There are several Spark on Kubernetes features that are currently being incubated in a fork -
-[apache-spark-on-k8s/spark](https://github.com/apache-spark-on-k8s/spark), which are expected to eventually make it into
-future versions of the spark-kubernetes integration.
+There are several Spark on Kubernetes features that are currently being worked on or planned to be worked on. Those features are expected to eventually make it into future versions of the spark-kubernetes integration.
```
[the remark this answers is not in the document]
Comment (liyinan926) [reply docs/running-on-kubernetes.md]: You meant removing the entire `Future Work` section?

File: docs/running-on-kubernetes.md [hunk 2]
```diff
@@ -185,6 +185,35 @@ To use a secret through an environment variable use the following options to the
 --conf spark.kubernetes.executor.secretKeyRef.ENV_NAME=name:key
 ```
 
+## Using Kubernetes Volumes
+Starting Spark 2.4.0, users can mount the following types of Kubernetes [volumes](https://kubernetes.io/docs/concepts/storage/volumes/) into the driver and executor pods:
```
[the remark this answers is not in the document]
Comment (liyinan926) [reply docs/running-on-kubernetes.md]: Done.

File: docs/running-on-kubernetes.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (srowen) [reply docs/running-on-kubernetes.md]: Just these two sentences, not the whole section. I figured it goes without saying there are features that may be worked on the the future that may go into Spark. But I don't feel strongly about it; this is fine.

**Review by srowen — APPROVED** [review]

**Comment by liyinan926** [comment]
Can this be merged?

**Comment by srowen** [comment]
Yeah, was just seeing if there were more comments. Merged to master

**Pull request closed without merging**